src/prng/chacha.rs

```diff
@@ -32,26 +32,26 @@ const STATE_WORDS: usize = 16;
 /// provided by auto-vectorisation.
 ///
 /// With the ChaCha algorithm it is possible to choose the number of rounds the
-/// core algorithm should run. By default `ChaChaRng` is created as ChaCha20,
-/// which means 20 rounds. The number of rounds is a tradeoff between performance
-/// and security, 8 rounds are considered the minimum to be secure. A different
-/// number of rounds can be set using [`set_rounds`].
+/// core algorithm should run. The number of rounds is a tradeoff between
+/// performance and security, where 8 rounds is the minimum potentially
+/// secure configuration, and 20 rounds is widely used as a conservative choice.
+/// We use 20 rounds in this implementation, but hope to allow type-level
+/// configuration in the future.
 ///
-/// We deviate slightly from the ChaCha specification regarding the nonce and
-/// the counter. Instead of a 64-bit nonce and 64-bit counter (or a 96-bit nonce
-/// and 32-bit counter in the IETF variant [3]), we use a 128-bit counter. This
-/// is because a nonce does not give a meaningful advantage for ChaCha when used
-/// as an RNG. The modification is provably as strong as the original cipher,
-/// though, since any distinguishing attack on our variant also works against
-/// ChaCha with a chosen nonce.
+/// We use a 64-bit counter and 64-bit stream identifier as in Benstein's
+/// implementation [1] except that we use a stream identifier in place of a
+/// nonce. A 64-bit counter over 64-byte (16 word) blocks allows 1 ZiB of output
+/// before cycling, and the stream identifier allows 2<sup>64</sup> unique
+/// streams of output per seed. Both counter and stream are initialized to zero
+/// but may be set via [`set_word_pos`] and [`set_stream`].
```

Comment:
Can you document that `set_word_pos` is only available with the `nightly` or `i128_support` features?

Reply:
I guess, but `i128` support will be standard at some point in the near-ish future anyway.

Reply:
Since a `u64` can represent 64 Exabytes, it feels a bit stupid using a `u128` here — should we use a `u64` instead and panic if the upper four bits of the counter are non-zero? This will suit the *vast* majority of users just fine, with only half the output size.